### PR TITLE
Implementation of GetUsers view, viewmodel and testing

### LINKED ISSUE #106: GetUsers view
I noticed that GetUsers and GetProjects display the user id and project id now. Please remove those from the respective views, as it should not be displayed to the user.

### PR DESCRIPTION
**Pull request by Pixeldom4**
1. **Addition of UsersPanel Class:**
Created a new UsersPanel class to display user information in a tabular format.
The panel is initialized with columns for user details like First Name, Last Name, Email, Tags, Desired Compensation, and Is Owner.

2. **Integration of UsersPanel into MyProjectsPanel:**
Added a JButton to MyProjectsPanel to fetch and display user information.
Included a UsersPanel instance in MyProjectsPanel.
Updated the actionPerformed method to handle the button click event, fetch user data, and display it in a new frame containing the UsersPanel.

3. **Enhancements to MyProjectsPanelViewModel:**
Added methods to handle user data (setUsersData) and to store and retrieve the selected project ID.

4. **Adjustments to the MyProjectsPanel Layout and Behavior:**
Modified the addProjects method to include a hidden project ID column, essential for internal logic but not visible to users.
Updated property change handling to manage new user data updates and project selections.

5. **New Unit Tests for MyProjectsPanel:**
Added tests to verify the new functionality, including handling user data updates, displaying user information, and managing project data.

### CODE REVIEW

**Comment by Pixeldom4** [comment]
I've made adjustments based on Rawq's [suggestions](https://github.com/sherwinokhowat/Group-Formation-System/pull/132#issue-2449765299). Thank you for the feedback!

File: src/test/java/view/MyProjectsPanelTest.java [hunk 1]
```diff
@@ -1,191 +1,129 @@
...
+//        myProjectsPanel.propertyChange(new PropertyChangeEvent(this, "dataUpdate", null, projectDataSet));
+//        JTable infoTable = (JTable) ((JScrollPane) myProjectsPanel.getComponent(0)).getViewport().getView();
+//
+//        // Assert
+//        assertEquals(2, infoTable.getRowCount());
+//        assertEquals("Title1", infoTable.getValueAt(0, 1));
+//        assertEquals("Description1", infoTable.getValueAt(0, 2));
+//        assertEquals("Yes", infoTable.getValueAt(0, 3));
+//        assertEquals("Edit", infoTable.getValueAt(0, 4));
+//    }
```
Comment (Raqkkkkk) [inline src/test/java/view/MyProjectsPanelTest.java]: remove commented code

File: src/main/java/view/MyProjectsPanel.java [hunk 2]
```diff
@@ -150,11 +190,29 @@ public void propertyChange(PropertyChangeEvent evt) {
...
-        // No implementation needed
+        if (e.getSource() == getUsersButton) {
+            // Retrieve the selected project ID
+            int projectId = myProjectsPanelViewModel.getSelectedProjectId();
+            getUsersController.getUsers(projectId);
+
+            // Display usersPanel in a new JFrame. Note that table size is large to accommodate large names/ groups of ppl
+            JFrame usersFrame = new JFrame("Users");
+            usersFrame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+            usersFrame.setSize(800, 600);
```
Comment (Raqkkkkk) [inline src/main/java/view/MyProjectsPanel.java]: can set to a smaller size

**Review by Raqkkkkk — CHANGES_REQUESTED** [review]
I think that the get users button can be put into the table and have one for each project rather than having it on the bottom

**Review by sherwinokhowat — CHANGES_REQUESTED** [review]
Everything looks good, but it would be nice if you delete the commented out code if it is not used.

**Review by Raqkkkkk — APPROVED** [review]
new commits look nice :D

**Review by sherwinokhowat — APPROVED** [review]
W changes :D

**Pull request merged**